src/app/modules/form/public-form/public-form.middlewares.ts

```diff
@@ -1,10 +1,40 @@
 import { RequestHandler } from 'express'
 import { StatusCodes } from 'http-status-codes'
+import mongoose from 'mongoose'
 
 import { WithForm } from '../../../../types'
+import getSubmissionModel from '../../../models/submission.server.model'
 import { FormDeletedError } from '../form.errors'
 import { isFormPublic } from '../form.service'
 
+const SubmissionModel = getSubmissionModel(mongoose)
+
+/**
+ * Express middleware function that checks if a form has exceeded its submission limits before allowing
+ * downstream middleware to handle the request. Otherwise, it returns a HTTP 404.
+ */
+export const isFormUnderSubmissionLimit: RequestHandler = (req, res, next) => {
+  const { form } = req as WithForm<typeof req>
+  if (form.submissionLimit !== -1) {
+    return SubmissionModel.countDocuments({ form: form._id })
+      .exec()
+      .then((queryResults) => {
+        if (queryResults >= form.submissionLimit) {
+          return res.status(StatusCodes.NOT_FOUND).json({
+            message:
+              'This form has reached the maximum number of submissions. If you think this is a mistake, please contact the agency that gave you the form link.',
+            isPageFound: true,
+            formTitle: form.title,
+          })
+        } else {
+          return next()
+        }
+      })
+  } else {
+    return next()
+  }
+}
```

Comment:
I think we can save all this code if the implementation simply deactivates the form after each submission